src/auth.rs

```diff
@@ -1,216 +0,0 @@
-
-#[derive(Debug, Error)]
-pub enum AuthError {
-    /// The login process was aborted from the server side.
-    #[error("Login was aborted.")]
-    LoginAborted,
-
-    /// The login process failed from the server side with an error message.
-    #[error("Login failed.")]
-    LoginFailed(String),
-
-    /// The server returned an unexpected gRPC status.
-    #[error("Error while making a request to the server.")]
-    ServerError(#[from] tonic::Status),
-
-    /// The auth URI generated by the SDK was unparseable.
-    #[error("Auth URI was invalid.")]
-    BadUri(#[source] InvalidUri),
-
-    /// An error occurred while trying to connect to the server.
-    #[error("Unable to connect to server.")]
-    Connection(#[source] tonic::transport::Error),
-
-    /// The action passed to [`login`] returned an error.
-    #[error("The login handler returned an error.")]
-    ActionError(#[source] Box<dyn std::error::Error + Send + Sync>),
-}
-
-impl From<AuthError> for MomentoError {
-    fn from(err: AuthError) -> Self {
-        match err {
-            AuthError::LoginAborted => MomentoError {
-                message: "aborted login".into(),
-                error_code: MomentoErrorCode::CancelledError,
-                inner_error: Some(err.into()),
-                details: None,
-            },
-            AuthError::LoginFailed(_) => MomentoError {
-                message: "login failed".into(),
-                error_code: MomentoErrorCode::PermissionError,
-                inner_error: Some(err.into()),
-                details: None,
-            },
-            AuthError::BadUri(_) => MomentoError {
-                message: "bad uri".into(),
-                error_code: MomentoErrorCode::BadRequestError,
-                inner_error: Some(err.into()),
-                details: None,
-            },
-            AuthError::Connection(_) => MomentoError {
-                message: "connection failed".into(),
-                error_code: MomentoErrorCode::InternalServerError,
-                inner_error: Some(err.into()),
-                details: None,
-            },
-            AuthError::ServerError(_) => MomentoError {
-                message: "server error".into(),
-                error_code: MomentoErrorCode::InternalServerError,
-                inner_error: Some(err.into()),
-                details: None,
-            },
-            AuthError::ActionError(_) => MomentoError {
-                message: "login action failed".into(),
-                error_code: MomentoErrorCode::UnknownError,
-                inner_error: Some(err.into()),
-                details: None,
-            },
-        }
-    }
-}
-
-impl From<ChannelConnectError> for AuthError {
-    fn from(value: ChannelConnectError) -> Self {
-        match value {
-            ChannelConnectError::BadUri(err) => Self::BadUri(err),
-            ChannelConnectError::Connection(err) => Self::Connection(err),
-        }
-    }
-}
-
-/// Initiate a login workflow.
```

Comment:
I feel this code belongs in the CLI as opposed to here. The CLI is the sole consumer of this. If we believe that it has broader utility and want to support it with our 1.0 guarantees I can add it back here.